src/base/base_uses.f90

```diff
@@ -14,7 +14,7 @@
 
 ! Check for OpenMP early on - ideally before the compiler fails with a cryptic message.
 #if !defined(_OPENMP)
-   "OpenMP is required. Please add -fopenmp to your Fortran compiler flags."
+#error "OpenMP is required. Please add the corresponding flag (eg. -fopenmp for GFortran) to your Fortran compiler flags."
```

Comment:
Unfortunately, `#error` is not support by the traditional preprocessor and therefore not available in Fortran (according to the standard).

Reply:
We use it already [here](https://github.com/cp2k/cp2k/blob/ec1cf9238dadae8274d74a47306a84345b4bbf05/src/environment.F#L231) (and in pw_cuda). Concerning the standard, CCE wants it. Most of the FPP compiler support it (expect [NAG](https://stackoverflow.com/questions/29609305/user-defined-errors-with-nagfor-preprocessor)?)

Reply:
We need something which breaks compilation, but there's no standard way of doing it. Usually this would be the job of the build system (once switched to CMake this will be solved in a portable way).
The only case we could hit are broken preprocessors, namely those which first parse for invalid preprocessor directives before checking whether the specific line is enabled or not, right?

Reply:
not the vanilla FPP for sure... CCE, GNU and Intel recognizes `#error`. I just want to have a meaningful message, I can even avoid it...

Reply:
Maybe we could move the error to a C file? For example there was a plan to move some of the code in [machine.F](https://github.com/cp2k/cp2k/blob/master/src/base/machine.F) to C.

Reply:
A C file? I think here we want to check that the Fortran compiler has the OpenMP part.
Currently, CCE gives the following error:
```
ftn-855 crayftn: ERROR MEMORY_UTILITIES, File = ../../../../../home/users/alazzaro/cp2k/CP2K_AUTO_REGTEST/CRAY-XC40-cce/popt/cp2k/src/common/memory_utilities.F, Line = 14, Column = 8 
  The compiler has detected errors in module "MEMORY_UTILITIES".  No module information file will be created for this module.


ftn-100 crayftn: ERROR MEMORY_UTILITIES, File = ../base/base_uses.f90, Line = 17, Column = 4 
  This statement must begin with a label, a keyword, or an identifier.

```

Then, I have to look at the specific line of the code to check the error...
Let's leave it without the directive, I will just change the message...

Reply:
You're right. I didn't think this through. I actually already added a check on the C [side](https://github.com/cp2k/cp2k/blob/master/src/grid/common/grid_library.c#L36).